buildrunner/docker/runner.py

```diff
@@ -251,6 +255,95 @@ def cleanup(self):
 
         self.container = None
 
+    def restore_caches(self, caches: OrderedDict, cache_archive_ext: str):  # pylint: disable=too-many-branches,too-many-locals
+        """
+        Restores caches from the host system to the destination location in the docker container.
+        """
+        if caches is None or not isinstance(caches, OrderedDict):
+            print("WARNING: caches is not an OrderedDict")
+            sys.exit()
```

Comment:
Instead of exiting in this case, what about raising an exception if this is bad input?

Reply:
I really try to avoid using sys.exit unless I'm in a simple command line script, personally.